aea/protocols/fipa/dialogues.py

```diff
@@ -263,7 +290,9 @@ def create_opponent_initiated(self, dialogue_opponent_pbk: Address, dialogue_id:
 
         :return: the created dialogue
         """
-        dialogue_label = DialogueLabel(dialogue_id, dialogue_opponent_pbk, dialogue_opponent_pbk)
+        assert dialogue_reference[0] != '' and dialogue_reference[1] == '', "Cannot initiate dialogue with preassigned dialogue_responder_reference!"
```

Comment:
That is (one of the) potentially harmful assertion, because a malicious opponent can easily make the skill fail.

Alternative is raise an exception that must be handled (or ignored) by the caller function.

Reply:
It is OK to leave as it is for now. We could add a `TODO` to don't forget that.

Reply:
Good point, the fix is a bit more involved so I will merge now and open an issue!